src/System.Management.Automation/FormatAndOutput/common/ComplexWriter.cs

```diff
@@ -353,27 +355,62 @@ static StringManipulationHelper()
         private static IEnumerable<GetWordsResult> GetWords(string s)
         {
             StringBuilder sb = new StringBuilder();
-            GetWordsResult result = new GetWordsResult();
+            StringBuilder vtSeqs = null;
+            Dictionary<int, int> vtRanges = null;
 
+            var valueStrDec = new ValueStringDecorated(s);
+            if (valueStrDec.IsDecorated)
+            {
+                vtSeqs = new StringBuilder();
+                vtRanges = new Dictionary<int, int>();
+                foreach (Match match in ValueStringDecorated.AnsiRegex.Matches(s))
+                {
+                    vtRanges.Add(match.Index, match.Length);
+                }
+            }
+
+            bool wordHasVtSeqs = false;
             for (int i = 0; i < s.Length; i++)
             {
-                // Soft hyphen = \u00AD - Should break, and add a hyphen if needed. If not needed for a break, hyphen should be absent
-                if (s[i] == ' ' || s[i] == '\t' || s[i] == s_softHyphen)
+                if (vtRanges?.TryGetValue(i, out int len) == true)
```

Comment:
This is used some times too but I am not sure we can move this in ValueStringDecorated.

```suggestion
                if (vtRanges?.TryGetValue(i, out int len))
```

Reply:
I don't think this check plus the if block should be moved to a utility method, as the logic is very specific to the calling context.
Also, the `== true` is required as `vtRanges` could be null.